client_lib/src/public/debug.c

```
@@ -59,12 +44,36 @@ void core_debug_addObjectInfo(const t_obj *obj, const char *info)
 	}
 }
+
+void core_debug_addObjectPathStep(const t_obj *unit, t_pos pos)
+{
+	if (!unit) return;
+
+	t_debug_entry *entry = core_static_findOrCreateEntry(unit->id);
+
+	// Create new path node
+	t_debug_path_node *new_node = malloc(sizeof(t_debug_path_node));
+	new_node->pos = pos;
```

Cppcheck (2.19.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)